perception/point_cloud_object_detection_and_recognition/tools/labeling_tool/label_model.cc

```diff
@@ -0,0 +1,403 @@
+        Eigen::Vector2d p(point.x, point.y);
+
+        p = p - pos;
+        a = (rot * a);
+        b = (rot * b);
+        d = (rot * d);
+
+        const Eigen::Vector2d ab = b - a;
+        const Eigen::Vector2d ad = d - a;
+        const Eigen::Vector2d ap = p - a;
+
+        const double ab_ap = ab.dot(ap);
+        const double ab_ab = ab.dot(ab);
+        const double ad_ap = ad.dot(ap);
+        const double ad_ad = ad.dot(ad);
+
+        if (0 < ab_ap && ab_ap < ab_ab && 0 < ad_ap && ad_ap < ad_ad)
+        {
+            return (int)(i);
+        }
+    }
+    return -1;
+}
+
+}  // anonymous namespace
+
+
+
+rosbag::View make_view() {}
+label_model::label_model(int argc, char* argv[], std::shared_ptr<id_to_labeled_object> object_map_ptr)
+    : object_map_ptr_(object_map_ptr)
+{
+    // Setup ROS
+    ros::init(argc, argv, "point_cloud_object_detector");
+    ros::Time::init();
+
+    // Set up node handler
+    auto nh = ros::NodeHandle(ros::this_node::getName());
+    nh_ = std::make_shared<ros::NodeHandle>(nh);
+
+    // Get bag from command-line
+    input_bag_name_ = argv[1];
+    output_bag_name_ = argv[2];
+
+    bag_.open(input_bag_name_, rosbag::bagmode::Read);
+
+    // This doesn't take into account camera distortion - for this application, it is ok, since we aren't using
+    // fisheyes and it's not a performance based application.
+    std::vector<std::string> image_topics = {"/odom", "/stereo/right/image_raw"};
+
+    std::vector<std::string> odom_topics = {"/odom"};
+    rosbag::View view_odom(bag_, rosbag::TopicQuery(odom_topics));
+    last_odom_ = view_odom.begin()->instantiate<nav_msgs::Odometry>();
+    v_ = std::make_shared<rosbag::View>(bag_, rosbag::TopicQuery(image_topics));
+
+    bag_iter_ = v_->begin();
+}
+
+void label_model::populate_map()
+{
+    tf2_ros::Buffer tfBuffer(ros::Duration(10.0));
+    tf2_ros::TransformListener tf_listener(tfBuffer);
+    geometry_msgs::TransformStamped g_velodyne_to_baselink;
+    geometry_msgs::TransformStamped g_baselink_to_stereoright;
+
+    ROS_INFO_STREAM("Getting the proper transforms");
+
+    while (true)
+    {
+        try
+        {
+            g_velodyne_to_baselink = tfBuffer.lookupTransform("base_link", "velodyne", ros::Time(0));
+            g_baselink_to_stereoright = tfBuffer.lookupTransform("stereo_right_cam", "base_link", ros::Time(0));
+
+            break;
+        }
+        catch (tf2::TransformException& ex)
+        {
+        }
+    }
+
+    pcodar::pcodar_params params;
+
+    //TODO(tbianchi) : Fix this lol
+    intrinsics_ << 704, 0, 476, 0, 0, 701, 300, 0, 0, 0, 1, 0;
+    tf::transformMsgToEigen(g_velodyne_to_baselink.transform, e_velodyne_to_baselink_);
+    tf::transformMsgToEigen(g_baselink_to_stereoright.transform, e_baselink_to_stereoright_);
+
+    ROS_INFO_STREAM("Done getting the proper transforms");
+
+    std::vector<std::string> topics;
+    topics.push_back(std::string("/velodyne_points"));
```

Comment:
Would be nice if this was a param or generic like ```lidar``` and ```camera``` so the command can be run with remaps on these topics